packages/client/lib/AuthorizationDetailsBuilder.ts

```diff
@@ -36,9 +35,9 @@ export class AuthorizationDetailsBuilder {
     return this;
   }
 
-  build(): AuthorizationDetails {
+  build(): CommonAuthorizationDetails {
```

Comment:
Hmzz, I also would like to see specific types being build. Otherwise the user will have to cast certain cases